eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/io/AccessionWriter.java

```diff
@@ -39,6 +47,36 @@ public AccessionWriter(SubmittedVariantAccessioningService service, AccessionRep
     public void write(List<? extends ISubmittedVariant> variants) throws Exception {
         List<AccessionWrapper<ISubmittedVariant, String, Long>> accessions = service.getOrCreateAccessions(variants);
         accessionReportWriter.write(accessions);
+        assertCountsMatch(variants, accessions);
+    }
+
+    void assertCountsMatch(List<? extends ISubmittedVariant> variants,
+                                   List<AccessionWrapper<ISubmittedVariant, String, Long>> accessions) {
+        if (variants.size() != accessions.size()) {
+            Set<ISubmittedVariant> accessionedVariants = accessions.stream()
+                                                                   .map(AccessionWrapper::getData)
+                                                                   .collect(Collectors.toSet());
+            HashSet<ISubmittedVariant> distinctVariants = new HashSet<>(variants);
+            int duplicateCount = variants.size() - distinctVariants.size();
+
+            List<ISubmittedVariant> variantsWithoutAccession = distinctVariants.stream()
+                                                                               .filter(v -> !accessionedVariants.contains(v))
+                                                                               .collect(Collectors.toList());
+            if (duplicateCount != 0) {
+                logger.warn("A variant chunk contains {} repeated variants. This is not an error, but please check " +
+                                    "it's expected.", duplicateCount);
+            }
+            if (variantsWithoutAccession.size() != 0) {
+                logger.error("A variant chunk wasn't accessioned properly. Only {} variants were accessioned out of " +
```

Comment:
this is a complicated message, but I guess that is not going to appear too much, and when it does we want as much detail as possible, so I think this is actually OK.

Reply:
A suggestion to to summarize the report a bit more while (I think) keeping it understandable:

> A problem occurred while accessioning a chunk. Size = {variants.size}, distinct = {distinctVariants.size}, duplicate = {duplicateCount}, accessioned = {accessionedVariants.size}, not accessioned = {variantsWithoutAccession.size}
> 
> The non-accessioned variants are: {variantsWithoutAccession.toString}

It allows to grep easily only the counts, or both the counts + list of variants, which is in the next line to make the formatting a bit nicer.